Yafc/Windows/PreferencesScreen.cs

```diff
@@ -94,15 +94,27 @@ public class PreferencesScreen : PseudoScreen {
             }
         }
 
-        /// <summary>Add a GUI element that opens a popup to allow the user to choose from the <paramref name="list"/>, which triggers <paramref name="select"/>.</summary>
+        /// <summary>Add a GUI element that opens a popup to allow the user to choose from the <paramref name="list"/>, which triggers <paramref name="selectItem"/>.</summary>
         /// <param name="text">Label to show.</param>
         /// <param name="width">Width of the popup. Make sure it is wide enough to fit text!</param>
-        /// <param name="allowNone">Whether to show a "Clear" option which sets the value to <c>null</c>.</param>
-        private void ChoiceObject<T>(ImGui gui, string text, T[] list, T current, Action<T> select, float width = 20f, bool allowNone = false) where T : FactorioObject {
+        private static void ChooseObject<T>(ImGui gui, string text, T[] list, T? current, Action<T> selectItem, float width = 20f) where T : FactorioObject {
             using (gui.EnterRow()) {
                 gui.BuildText(text, topOffset: 0.5f);
                 if (gui.BuildFactorioObjectButtonWithText(current)) {
-                    gui.BuildObjectSelectDropDown(list, DataUtils.DefaultOrdering, select, text, width: width, allowNone: allowNone);
+                    gui.BuildObjectSelectDropDown(list, DataUtils.DefaultOrdering, selectItem, text, width: width);
+                }
+            }
+        }
+
+        /// <summary>Add a GUI element that opens a popup to allow the user to choose from the <paramref name="list"/>, which triggers <paramref name="selectItem"/>.
+        /// An additional "clear" or "none" option will also be displayed.</summary>
+        /// <param name="text">Label to show.</param>
+        /// <param name="width">Width of the popup. Make sure it is wide enough to fit text!</param>
+        private static void ChooseObjectWithNone<T>(ImGui gui, string text, T[] list, T? current, Action<T?> selectItem, float width = 20f) where T : FactorioObject {
```

Comment:
Here, I separated `ChooseObject` and `ChooseObjectWithNone` (along with creating several other `...WithNone` methods) so the compiler could check the `selectItem` argument. If the UI is displaying a 'none' or 'clear' item, `selectItem` must accept a null value. Nullable annotations don't have the ability to tell the compiler about that with an allowNone parameter, so I moved that into the method name instead.